### PR TITLE
Update cart api to use the redux toolkit

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes
Preparatory branch to get the cart ready for the copy/move components

## Related issue
N/A

## Checklist
N/A

### CODE REVIEW

File: src/main/webapp/resources/js/pages/cart/components/Cart.jsx [hunk 1]
```diff
@@ -1,23 +1,23 @@
-import React, { lazy, Suspense, useEffect, useState } from "react";
-import { connect } from "react-redux";
-import PropTypes from "prop-types";
 import { Layout } from "antd";
-import { actions } from "../../../redux/reducers/cart";
+import React, { lazy, Suspense, useState } from "react";
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/cart/components/Cart.jsx:2]: Remove `useState` and use React.useState below instead

File: src/main/webapp/resources/js/pages/cart/components/Cart.jsx [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
 
-  useEffect(() => {
-    setCollapsed(count === 0);
-    if (count !== 0) {
-      loadCart();
-    }
-  }, [count]);
+  const { data: count = 0 } = useCountQuery();
+
+  // useEffect(() => {
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/cart/components/Cart.jsx:15]: Remove commented out code if its not required here and below

File: src/main/webapp/resources/js/pages/cart/components/CartSamples.jsx [hunk 3]
```diff
@@ -65,17 +68,40 @@ const EmptyCartButton = styled(Button)`
...
+  } = useGetCartQuery();
+  const [emptyCart] = useEmptyMutation();
+  const [removeSample] = useRemoveSampleMutation();
+  const [removeProject] = useRemoveProjectMutation();
+
+  React.useEffect(() => {
+    setSamples(allSamples);
+  }, [allSamples]);
+
+  const filterSamples = (e) => {
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/cart/components/CartSamples.jsx:87]: Change parameter `e` to something more clear

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Just a few changes/questions below for now to start :)

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UICartService.java [hunk 4]
```diff
@@ -188,11 +188,10 @@ public CartUpdateResponse removeProject(Long id, Locale locale) {
 	/**
 	 * Get a list of sample in the cart belonging to a list of projects
 	 *
-	 * @param projectIds List of identifiers for project to get the samples for.
 	 * @return {@link List} of {@link CartProjectModel}s containing project and sample information for items in the cart.
 	 */
-	public List<CartProjectModel> getSamplesForProjects(List<Long> projectIds) {
-		List<Project> projects = (List<Project>) projectService.readMultiple(projectIds);
+	public List<CartProjectModel> getSamplesForProjects() {
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UICartService.java:193]: Updated in 6fc9d109937ecb2f018605e6096ec2db568ca560

File: src/main/webapp/resources/js/apis/cart/cart.js [hunk 5]
```diff
@@ -6,26 +6,92 @@ import { setBaseUrl } from "../../utilities/url-utilities";
...
+    }),
+    removeProject: build.mutation({
+      query: ({ id }) => ({
+        url: "/project",
+        method: "DELETE",
+        params: { id },
+      }),
+      invalidatesTags: [{ type: "Samples", id: "LIST" }, "CartCount"],
+      transformResponse: (response) => {
+        console.log(response);
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/apis/cart/cart.js:65]: Updated in ae8cf81af2ce05f0a1ccaa36905d97735fd87eb2

File: src/main/webapp/resources/js/pages/cart/components/Cart.jsx (hunk 1 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/cart/components/Cart.jsx:2]: Good catch, updated in 4bf7933335a355a16f1c1a4f4eb34b66014b2d76

File: src/main/webapp/resources/js/pages/cart/components/Cart.jsx (hunk 2 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/cart/components/Cart.jsx:15]: Updated in 4bf7933335a355a16f1c1a4f4eb34b66014b2d76

File: src/main/webapp/resources/js/pages/cart/components/CartSamples.jsx (hunk 3 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/cart/components/CartSamples.jsx]: Updated in b8d4c029682778d53d9631a7e3892abf3614c3ea

File: src/main/webapp/resources/js/pages/cart/components/CartSamples.jsx [hunk 6]
```diff
@@ -1,16 +1,19 @@
+import { Button, Empty, Input, notification, Space, Spin } from "antd";
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/cart/components/CartSamples.jsx:1]: File comments

File: src/main/webapp/resources/js/pages/cart/components/share/index.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { Row } from "antd";
+import React from "react";
+
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/cart/components/share/index.js:3]: Add a file comment for this

File: src/main/webapp/resources/js/pages/cart/components/CartSamples.jsx (hunk 6 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/cart/components/CartSamples.jsx:1]: Added in 30d93f2b20e775f8f08280946cda54c892eb7d20

File: src/main/webapp/resources/js/pages/cart/components/share/index.js (hunk 7 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/cart/components/share/index.js:3]: Added in 30d93f2b20e775f8f08280946cda54c892eb7d20

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**